scripts/ci.ps1

```diff
@@ -0,0 +1,69 @@
+param(
+    # CMake Version to test with
+    [Parameter()]
+    [string]
+    $CMakeVersion = "3.10.0"
+)
+$ErrorActionPreference = "Stop"
+
+# The root directory of our repository:
+$REPO_DIR = Split-Path (Split-Path $MyInvocation.MyCommand.Source -Parent) -Parent
+$SCRIPTS_DIR = Join-Path $REPO_DIR "scripts"
+
+. $SCRIPTS_DIR/util.ps1
+
+Write-Debug "Repository directory is $REPO_DIR"
+
+$cmake_binary = Install-TestCMake -Version $CMakeVersion
+ConvertTo-Json $cmake_binary
+Write-Host "cmake $cmake_binary"
+
+Write-Debug "Preparing test utilities..."
+& $SCRIPTS_DIR/prepare-test.ps1 -CMakePath $cmake_binary
+
+$bindir = Join-Path $REPO_DIR ".ci-build"
+
+& $cmake_binary "-H$REPO_DIR" "-B$bindir"
+$retc = $LASTEXITCODE
```

Comment:
Probably hold-over bash-ism where you have to grab the exit code immediately.

I wasn't seeing PS throw on cmake configure/build failures. It happily continued without checking the exit code or output. I might wrap these commands in a function that checks their results rather than this boilerplate incantation, unless I can find a way to make it respect the exit code.